cypress-tests/cypress/pageobjects/ide/ProjectTree.ts

```diff
@@ -136,5 +159,64 @@ export class ProjectTree {
         this.clickOnItem(`${pathToItem}/${fileName}`)
     }
 
+    waitProjectImported(projectName: string, rootSubitem: string) {
+        cy.log("**=> ProjectTree.waitProjectImported**")
+            .then(() => {
+                let attempts: number = 4;
+                let pollingEvery: number = 30000;
+                let currentAttempt: number = 1;
+
+                this.waitImported(projectName, rootSubitem, attempts, currentAttempt, pollingEvery)
+            })
+    }
+
+    private waitImported(projectName: string, rootSubitem: string, attempts: number, currentAttempt: number, pollingEvery: number): Promise<void> {
+        return new Promise((resolve, reject) => {
+            let rootItem: string = `/${projectName}`;
+
+            this.expandItem(rootItem)
+            this.waitItemExpanded(rootItem)
+
+            cy.get('body')
+                .then(body => {
+                    let rootItemLocator: string = this.getTreeItemLocator(`/${projectName}`);
```

Comment:
What about extracting enclosed code into the external method to simplify readability and make it possible to reuse it?